### PR TITLE
chore: add snappcloud to adopters

### PR DESCRIPTION
**Pull request by m-yosefpor**
Hello from SnappCloud!

We at SnappCloud are excited to share our journey with Contour and would like to be included in the list of Contour Adopters. We have successfully integrated Contour into our infrastructure and actively contribute to the open-source community by developing solutions around Contour. Our contributions include several tools and plugins enhancing Contour's functionality. We believe our experience and contributions would be a valuable addition to the Contour Adopters page.

Thank you for considering our request.

### CODE REVIEW

**Review by davinci26 — APPROVED** [review]
fantastic

File: site/content/resources/adopters.md
Comment (davinci26) [inline site/content/resources/adopters.md:65]: @projectcontour/maintainers we are also  finding the need to build this.

Should we bring in an implementation of this into `projecontour` organization? Should we open an issue for that?

Comment (skriss) [reply site/content/resources/adopters.md:65]: 🤔 we have some examples of using Gatekeeper for this -- see https://projectcontour.io/guides/gatekeeper/ and https://github.com/projectcontour/contour/tree/main/examples/gatekeeper. I think we would also prefer to make more use of CEL validation where possible, since they don't require deploying an additional component.

That said, if there is still a need for a custom admission webhook and is something that there is shared community interest in, we could definitely look at creating a repo for it in the projectcontour org.

**Review by skriss — APPROVED** [review]
Thanks @m-yosefpor, this is great!

**Review by skriss — DISMISSED** [review]
Just need to add DCO signoff to your commit per https://github.com/projectcontour/contour/pull/6056/checks?check_run_id=20191677028 -- `git commit --amend --signoff && git push -f` should do it.

File: site/content/resources/adopters.md
Comment (davinci26) [reply site/content/resources/adopters.md:65]: TIL, that is great

Comment (davinci26) [reply site/content/resources/adopters.md:65]: let me digest that but seems really useful!

**Comment by m-yosefpor** [comment]
> Just need to add DCO signoff to your commit per https://github.com/projectcontour/contour/pull/6056/checks?check_run_id=20191677028 -- `git commit --amend --signoff && git push -f` should do it.

done @skriss

File: site/content/resources/adopters.md
Comment (m-yosefpor) [reply site/content/resources/adopters.md:65]: Regarding CEL, AFAIK, Gatekeeper and CEL validation in CRDs are generally limited to the properties of the resource itself. So for HTTPProxy host FQDN conflicts, it's not possible to do so. It is possible to do it with other policy engines such as kyverno, as they support [apiCall](https://kyverno.io/docs/writing-policies/external-data-sources/#variables-from-kubernetes-api-server-calls), however it still requires to get all HTTPProxies in all namespaces with every update on HTTPProxies. We implemented a custom webhook, so we can cache all FQDNs to avoid such heavy lookups for every operation.

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]

File: site/content/resources/adopters.md
Comment (ssttehrani) [reply site/content/resources/adopters.md:65]: JFYI:
We're pleased to inform you about the release of the new version of our custom admission webhook. This update focuses on improving stability, making it a dependable choice for production environments. The admission webhook maintains its commitment to enforcing security standards for Contour in multi-tenant clusters.

**Pull request merged**
